Prompt window on clear files

midlan opened the pull request:
Sometimes I accidentally clicked on clear files button when I was selecting some file (since the clear button is really close to that selectbox). So I added prompt if user really wants to clear all files.

alpha0010 commented:
Although potentially useful, a confirmation dialog can quickly become both [annoying and and useless](https://alistapart.com/article/neveruseawarning).  Perhaps addressing this would be better by moving the clear button, or at least making it possible to disable the dialog?

midlan commented:
I don't think that would be annoying. Typically you delete files few times when debugging. Since the delete action can!t be undid, the prompt seems handy for me.

Danny3 commented:
For me this would be definitely annoying.
I hate it when the computer interrupts me just to tell me something that I already know.
I can only see this useful for new users only and even so the debug files can be recreated easily, I don't see anything important lost.
If this gets merged, please make it as opt-in in the config
Thank you!

midlan commented:
> the debug files can be recreated easily

This is definitely not true. I created this pull request just because I deleted myself files that **can't** be recreated easily.
Imagine PHP application, that use caching. After clearing cache only the first request will trace through all the code, second and later request will load from cache. S OK you deleted cache, catched few types of requests request (you are searching for bottlenck in mutiple pages) with profiler. Now you are going to view them in webgrind, but ouch you just misclicked the selectbox and deleted all your prepared request to debug. So you must invalidate cache and prepare all the requests one more time.

This is only easy example where cache invalidate is needed before catching request. Imagine much worser requests to prepare, eg. createing or deleting item in database etc.

Woet commented:
I've just submitted a pull request that addresses this issue: #144.

alpha0010 commented:
Closing in favor of #144.
